geomesa-dynamo/geomesa-dynamodb-datastore/src/main/scala/org/locationtech/geomesa/dynamodb/data/DynamoDBFeatureWriter.scala

```diff
@@ -0,0 +1,137 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.dynamodb.data
+
+import java.nio.charset.StandardCharsets
+import java.util.{Date, UUID}
+
+import com.amazonaws.services.dynamodbv2.document.spec.PutItemSpec
+import com.amazonaws.services.dynamodbv2.document.{Item, PrimaryKey, Table}
+import com.google.common.primitives.{Bytes, Ints, Longs}
+import com.vividsolutions.jts.geom.Point
+import org.geotools.data.simple.SimpleFeatureWriter
+import org.joda.time.DateTime
+import org.locationtech.geomesa.dynamo.core.DynamoPrimaryKey
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+import org.locationtech.geomesa.utils.text.WKBUtils
+import org.opengis.feature.`type`.AttributeDescriptor
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+
+trait DynamoDBPutter {
+  // Allows impl changes between appending and updating writers
+  protected def dynamoDBPut(t: Table, i: Item): Unit = ???
+}
+
+trait DynamoDBFeatureWriter extends SimpleFeatureWriter with DynamoDBPutter {
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+  def sft: SimpleFeatureType
+  def table: Table
+
+  private[this] var curFeature: SimpleFeature = null
+
+  val dtgIndex = sft.getDtgIndex.get
+
+  private val encoder = new KryoFeatureSerializer(sft)
+
+  private def serialize(item: Item, attr: AnyRef, desc: AttributeDescriptor) = {
+    import java.{lang => jl}
+    desc.getType.getBinding match {
+      case c if c.equals(classOf[jl.Boolean]) =>
+        item.withBoolean(desc.getLocalName, attr.asInstanceOf[jl.Boolean])
+
+      case c if c.equals(classOf[jl.Integer]) =>
+        item.withInt(desc.getLocalName, attr.asInstanceOf[jl.Integer])
+
+      case c if c.equals(classOf[jl.Double]) =>
+        item.withDouble(desc.getLocalName, attr.asInstanceOf[jl.Double])
+
+      case c if c.equals(classOf[String]) =>
+        item.withString(desc.getLocalName, attr.asInstanceOf[String])
+
+      case c if c.equals(classOf[java.util.Date]) =>
+        item.withLong(desc.getLocalName, attr.asInstanceOf[java.util.Date].getTime)
+
+      case c if classOf[Point].isAssignableFrom(c) =>
+        item.withBinary(desc.getLocalName, WKBUtils.write(attr.asInstanceOf[Point]))
```

Comment:
add a default case with a meaningful exception, otherwise we'll just get inscrutable scala.MatchErrors